### PR TITLE
virt-viewer: migrate to libsoup_3

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

As in the title.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/virtualization/libgovirt/default.nix [hunk 1]
```diff
@@ -36,9 +35,11 @@ stdenv.mkDerivation rec {
 
   propagatedBuildInputs = [
     glib
-    librest
+    librest_1_0
   ];
 
+  NIX_CFLAGS_COMPILE = lib.optionalString stdenv.isDarwin "-Wno-error";
```
[the remark this answers is not in the document]
Comment (Atemu) [reply pkgs/applications/virtualization/libgovirt/default.nix:41]: Not sure what this is about but upstream made the patch redundant.

**Review by Atemu — CHANGES_REQUESTED** [review]
I don't think it's a good idea to migrate to libsoup_3 when upstream hasn't completed the transition yet.

https://gitlab.com/virt-viewer/virt-viewer/-/issues/88

Unlike regular distros, we have the luxury of allowing multiple versions of libsoup in the system. I don't see why we need to patch that.

If there is a good reason, it needs to go in the commit message.

File: pkgs/applications/virtualization/libgovirt/default.nix [hunk 2]
```diff
@@ -36,9 +35,11 @@ stdenv.mkDerivation rec {
...
+
   enableParallelBuilding = true;
```
Comment (jtojnar) [inline pkgs/applications/virtualization/libgovirt/default.nix:43]: This is redundant with Meson.

**Comment by wegank** [comment]
Closing in favour of #214384.

**Pull request closed without merging**